subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/AbstractGradleBuildPerformanceTestRunner.groovy

```diff
@@ -83,10 +72,6 @@ abstract class AbstractGradleBuildPerformanceTestRunner<R extends PerformanceTes
     }
 
     protected void defaultSpec(BuildExperimentSpec.Builder builder) {
-        builder.setListener(buildExperimentListener)
-        builder.setInvocationCustomizer(invocationCustomizer)
-        builder.setBuildMutators(buildMutators)
-        builder.setMeasuredBuildOperations(ImmutableList.copyOf(measuredBuildOperations))
     }
```

Comment:
Should this be `abstract` now?

Reply:
I think an empty implementation is fine. You don't need to override `defaultSpec` in your subclass if you don't need to prepare anything in the spec.

Reply:
You are now calling `super.defaultSpec()`, which is easy to forget. I think that's more error-prone than having to implement the method.

Reply:
Anyway, I don't feel very strongly about this.